flexget/plugins/clients/tarantool_queue.py

```diff
@@ -0,0 +1,145 @@
+from loguru import logger
+from flexget.entry import Entry
+from flexget.event import event
+from flexget import plugin
+
+logger = logger.bind(name='tarantool_queue')
+
+class TarantoolTube:
+    def __init__(self, queue, tube_name):
+        self._queue = queue
+        self._tube_name = tube_name
+
+    def put(self, data):
+        return self._queue.call('queue.tube.'+self._tube_name+':put', data)
+
+    def take(self, timeout=None):
+        return self._queue.call('queue.tube.'+self._tube_name+':take', timeout)
+
+    def ack(self, task_id):
+        return self._queue.call('queue.tube.' + self._tube_name + ':ack', task_id)
+
+class TarantoolQueue:
+    def __init__(self, tarantool_module, host, port, user, password):
+        self._server = tarantool_module.connect(host, port, user=user, password=password)
+        self._tubes = {}
+
+    def tube(self, tube_name):
+        if tube_name in self._tubes:
+            tube = self._tubes[tube_name]
+        else:
+            tube = TarantoolTube(self, tube_name)
+            self._tubes[tube_name] = tube
+        return tube
+
+    def call(self, function, *args):
+        return self._server.call(function, *args)
+
+class PluginTarantoolQueueBase:
+    schema = {
+        'anyOf': [
+            {'type': 'boolean'},
+            {
+                'type': 'object',
+                'properties': {
+                    'host': {'type': 'string'},
+                    'port': {'type': 'integer'},
+                    'user': {'type': 'string'},
+                    'password': {'type': 'string'},
+                    'tube': {'type': 'string'},
+                    'enabled': {'type': 'boolean'},
+                },
+                'additionalProperties': False,
+            },
+        ]
+    }
+
+    def __init__(self):
```

Comment:
Plugins should avoid `__init__`. We want them to be stateless. These can often "leak" memory. At minimum config should be kept out as it leads too easily into situation where reload does not apply.